src/borg/compress.pyx

```diff
@@ -179,12 +211,50 @@ class ZLIB(CompressorBase):
             raise DecompressionError(str(e)) from None
 
 
+class Auto(CompressorBase):
+    """
+    Meta-Compressor that decides which compression to use based on LZ4's ratio.
+
+    As a meta-Compressor the actual compression is deferred to other Compressors,
+    therefore this Compressor has no ID, no detect() and no decompress().
+    """
+
+    ID = None
+    name = 'auto'
+
+    logger = create_logger('borg.debug.file-compression')
+
+    def __init__(self, compressor):
+        super().__init__()
+        self.compressor = compressor
+        self.lz4 = get_compressor('lz4')
+        self.none = get_compressor('none')
+
+    def compress(self, data):
+        lz4_data = self.lz4.compress(data)
+        if len(lz4_data) < 0.97 * len(data):
+            return self.compressor.compress(data)
+        elif len(lz4_data) < len(data):
+            return lz4_data
```

Comment:
even though lz4 decompression is very fast, we could still use a minimum compression here, like:
```
len(lz4_data) < 0.999 * len(data)
or even
len(data) >= 100000 and len(lz4_data) < 0.999 * len(data)
```